week_1.py

```diff
@@ -0,0 +1,188 @@
+        print(f'The number of loyalty points you have are: {loyalty_points1}')
+
+    def calc_loyalty_discount(self): #calc prices using loyalty points
+        discount_price = round(self.total_price - (self.calc_loyalty_points() * 0.2))
+        print(f'The discount price using loyalty points is: £{discount_price}')
+
+# cr = CashRegister()
+# cr.add_item("t-shirt", 5)
+# cr.add_item("shoes", 10)
+# cr.add_item("socks", 15)
+# cr.show_items()
+# print("\n")
+# cr.remove_item("t-shirt")
+# cr.show_items()
+# print("\n")
+# cr.apply_discount(20, "socks")
+# print("\n")
+# cr.get_total()
+# print("\n")
+# cr.show_loyalty_points()
+# print("\n")
+# cr.calc_loyalty_discount()
+# print("\n")
+# cr.reset_register()
+
+
+# EXAMPLE code run:
+
+# ADD
+
+
+"""
+
+TASK 2
+
+Write a base class to represent a student. Below is a starter code. 
+Feel free to add any more new features to your class. 
+As a minimum a student has a name and age and a unique ID.
+
+Create a new subclass from student to represent a concrete student doing a specialization, for example:
+Software Student and Data Science student. 
+
+"""
+
+
+class Student:
+
+    def __init__(self, name, age, id):
+        self.name = name
+        self.age = age
+        self.id = id
+        self.subjects = dict() #{} #subject:grade
+
+class CFGStudent(Student):
+
+    def __init__(self, name, age, id, specialisation, attendance): #specialisation
+        self.name = name
+        self.age = age
+        self.id = id
+        self.specialisation = specialisation
+        self.attendance = attendance
+        Student.__init__(self, self.name, self.age, self.id) #only the attributes from above
+
+    def add_subject(self, subject, grade):
+        self.subjects[subject] = grade
+        return self.subjects
+
+    def remove_subject(self, subject):
+        self.subjects.pop(subject)
+        return self.subjects
+
+    def show_items(self): #show all the subjects and grades
+        for grades, subjects in self.subjects.items():
+            print(grades, subjects)
+
+    def show_subjects(self): #show subjects only
+        for subjects in self.subjects.keys():
+            print(subjects)
+
+    def calculate_grade(self): #calculate average grades
+        overall_grade = sum(self.subjects.values())
+        average_grade = round(overall_grade / len(self.subjects))
+        if average_grade > 70:
+            print(f'Congratulations you have achieved {average_grade} are a first class student!')
+        elif (average_grade > 60) and (average_grade < 71):
+            print(f'Fantastic you scored {average_grade} and achieved a 2:1!')
+        elif (average_grade > 50) and (average_grade < 61):
+            print(f'Well done you scored {average_grade} and achieved a 2:2!')
+        elif (average_grade > 40) and (average_grade < 51):
+            print(f'Good you you scored {average_grade} and passed!')
+        else:
+            print(f'Unfortunately you scored {average_grade} and failed :(')
```

Comment:
great addition of the extra printed output to see their grades classification not just the average grade!